### PR TITLE
Enable mirror linter and fix findings

### PR DESCRIPTION
**Pull request by jtraglia**
**What type of PR is this?**

Other

**What does this PR do? Why is it needed?**

This PR enables the [`mirror`](https://github.com/butuzov/mirror) linter which...

> suggests use of alternative functions/methods in order to gain performance boosts by avoiding unnecessary []byte/string conversion calls.

Here were the findings:

```
$ golangci-lint run
validator/rpc/auth_token.go:140:15: avoid allocations with (*bytes.Buffer).WriteString (mirror)
        if _, err := bytesBuf.Write([]byte(fmt.Sprintf("%x", jwtKey))); err != nil {
                     ^
validator/rpc/auth_token.go:143:15: avoid allocations with (*bytes.Buffer).WriteString (mirror)
        if _, err := bytesBuf.Write([]byte("\n")); err != nil {
                     ^
validator/rpc/auth_token.go:146:15: avoid allocations with (*bytes.Buffer).WriteString (mirror)
        if _, err := bytesBuf.Write([]byte(token)); err != nil {
                     ^
validator/rpc/handlers_keymanager.go:135:18: avoid allocations with bytes.NewBufferString (mirror)
                        ctx, s.valDB, bytes.NewBuffer([]byte(req.SlashingProtection)),
                                      ^
```

### CODE REVIEW

**Review by terencechain — DISMISSED** [review]

**Comment by jtraglia** [comment]
Looks like we'll need to use a newer version of golangci-lint. I'll fix that tomorrow.

**Comment by jtraglia** [comment]
This should be ready now.

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -54,7 +54,7 @@ jobs:
       - name: Golangci-lint
         uses: golangci/golangci-lint-action@v3
         with:
-          version: v1.52.2
+          version: latest
```
Comment (prestonvanloon) [inline .github/workflows/go.yml:57]: Can you pin to a specific version please? We've had pain in the past where this is suddenly incompatible and blocking all PRs

Comment (jtraglia) [reply .github/workflows/go.yml:57]: Yes, will update.

**Review by prestonvanloon — DISMISSED** [review]
Thank you!

**Comment by prestonvanloon** [comment]
Oh, conflicts in .golangci.yml :scream:

**Comment by jtraglia** [comment]
> Oh, conflicts in .golangci.yml 😱

Fixed!

**Review by terencechain — APPROVED** [review]

**Pull request merged**